theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -781,3 +783,77 @@ def test_compare_1D_and_2D_upsampling_values(self):
         f_1D = theano.function([], mat_1D, mode=self.compile_mode)
         f_2D = theano.function([], mat_2D, mode=self.compile_mode)
         utt.assert_allclose(f_1D(), f_2D(), rtol=1e-06)
+
+
+def test_conv2d_grad_wrt_inputs():
+
+    inputs_shapes = [(8, 1, 12, 12), (8, 1, 18, 18), (2, 1, 4, 4),
+                     (6, 1, 10, 11), (2, 1, 6, 5), (1, 5, 9, 9)]
+    filters_shapes = [(5, 1, 2, 2), (4, 1, 3, 3), (2, 1, 3, 3),
+                      (1, 1, 2, 5), (4, 1, 2, 2), (4, 5, 2, 2)]
+    subsamples = [(1, 1), (2, 2)]
+    border_modes = ["valid", "full"]
+    filter_flip = [True, False]
+    x = T.cast(T.ftensor4('x'), theano.config.floatX)
+    for (in_shape, fltr_shape) in zip(inputs_shapes, filters_shapes):
+        input_val = np.random.rand(*in_shape).astype(theano.config.floatX)
+
+        filter_val = np.random.rand(*fltr_shape).astype(theano.config.floatX)
+        for bm in border_modes:
+            for ss in subsamples:
+                for ff in filter_flip:
+                    cgi = conv2d_grad_wrt_inputs(output_grad=input_val,
+                                                 filters=filter_val,
+                                                 filter_shape=fltr_shape,
+                                                 input_shape=in_shape,
+                                                 border_mode=bm,
+                                                 subsample=ss,
+                                                 filter_flip=ff)
+
+                    conv_func = T.nnet.conv.conv2d(x,
+                                                   filters=filter_val,
+                                                   border_mode=bm,
+                                                   filter_shape=fltr_shape,
+                                                   image_shape=in_shape,
+                                                   subsample=ss)
+
+                    f_prime = theano.grad(conv_func.sum(), x)
+                    f = theano.function([x], f_prime)
+                    T.allclose(cgi, f(input_val))
```

Comment:
`T.allclose` here does not actually perform a comparison. It just builds a symbolic expression that represents the (approximate) equality between `cgi` (a symbolic variable) and `f(input_val)` (a constant).
What you want instead is another Theano function computing a value for `cgi` (given `output_grad` and `filters`), and calling `utt.assert_allclose(..., ...)` on the value returned by that function and by f (`f_prime` if you follow my code earlier).